app/models/user/user.ts

```diff
@@ -0,0 +1,39 @@
+import { Instance, SnapshotOut, types } from "mobx-state-tree"
+import auth from "@react-native-firebase/auth"
+import RevenueCat from "react-native-purchases"
+
+export const UserModel = types
+  .model("User")
+  .props({
+    currentUserId: types.maybe(types.string),
+  })
+  .views((self) => ({
+    get currentUser() {
+      return auth().currentUser
+    },
+  }))
+  .actions((self) => ({
+    setCurrentUserId(uid?: string) {
+      self.currentUserId = uid
+    },
+    async afterCreate() {
+      if (!self.currentUser) {
+        auth().signInAnonymously()
```

Comment:
Can you create users only upon purchasing?

Reply:
I think it's better to create an anonymous user for everyone. It'll provide us more flexibility in the future and possibly better analytics. It's also free so why not 😄